### PR TITLE
Enforce IDE0090 and fix remaining issues

### PR DESCRIPTION
**Pull request by Evangelink**

### CODE REVIEW

**Review by sharwell — APPROVED** [review]

File: .editorconfig [hunk 1]
```diff
@@ -234,6 +234,9 @@ dotnet_diagnostic.IDE0065.severity = silent
 # IDE0072: Populate switch
 dotnet_diagnostic.IDE0072.severity = silent
 
+# IDE0090: Use 'new(...)'
```
Comment (mavasani) [inline .editorconfig:237]: This is redundant- the style category entry above already bumps all IDE style diagnostics to be warning by default.

**Review by mavasani — COMMENTED** [review]
The changes are fine but the rule is still not enforced on build- you can try to push a violating commit and confirm that build still succeeds. Build enforcement needs us to move to 3.9.x code style package after VS2019 16.9 is released.

**Review by mavasani — APPROVED** [review]
I can merge once you revert the dummy violating commit.

**Comment by Evangelink** [comment]
Is it worth creating a ticket for the move to the `3.9.x` package?

**Comment by mavasani** [comment]
> Is it worth creating a ticket for the move to the 3.9.x package?

Sure, feel free to create one.

**Pull request merged**